start_admin_sshd.sh

```diff
@@ -21,8 +21,10 @@ get_user_data_keys() {
     # Extract the keys from user-data json
     local raw_keys
     local key_type="${1:?}"
-    if ! raw_keys=$(jq --arg key_type "${key_type}" -e -r '.["ssh"][$key_type][]' "${user_data}" 2>/dev/null); then
-      log "Failed to parse ${key_type} from ${user_data}"
+    # ? signifies an optional object identifier-index and doesn't return a 'failed to iterate' error
+    # in the event that a key_type is missing. If jq returns nothing, then that is caught by the -z.
+    if ! raw_keys=$(jq --arg key_type "${key_type}" -e -r '.["ssh"][$key_type][]?' "${user_data}") \
+    || [[ -z "${raw_keys}" ]]; then
```

Comment:
I'm pretty sure you can use the `-e` flag for that.

> -e / --exit-status:

           Sets  the  exit  status of jq to 0 if the last output values
           was neither false nor null, 1 if the last output  value  was
           either  false or null, or 4 if no valid result was ever pro‐
           duced. Normally jq exits with 2 if there was any usage prob‐
           lem  or  system  error,  3 if there was a jq program compile
           error, or 0 if the jq program ran.

```suggestion
    if ! raw_keys=$(jq -e --arg key_type "${key_type}" -e -r '.["ssh"][$key_type][]?' "${user_data}"); then
```

Reply:
When I was testing I found that jq actually will return a 0 on an empty list, since the list exists but just happens to not be populated.

Reply:
After some further testing, I verified the -z is not necessary.